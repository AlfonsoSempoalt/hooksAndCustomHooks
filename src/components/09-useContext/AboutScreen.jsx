import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutScreen = (props) => {
  const { user, setUser } = useContext(UserContext);
  const handleLogOut = () => {
    setUser({});
  };
  return (
    <>
      <h1>AboutScreen</h1>
      <hr />
      <pre>{JSON.stringify(user)}</pre>
      <button className='btn btn-warning' onClick={handleLogOut}>
        logout
      </button>
    </>
  );
};

export default AboutScreen;
