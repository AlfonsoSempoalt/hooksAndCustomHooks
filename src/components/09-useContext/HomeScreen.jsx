import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const HomeScreen = (props) => {
  const { user } = useContext(UserContext);
  console.log(user);
  return (
    <>
      <h1>Home page</h1>
      <hr />

      <pre>{ JSON.stringify( user )}</pre>
    </>
  );
};

export default HomeScreen;
