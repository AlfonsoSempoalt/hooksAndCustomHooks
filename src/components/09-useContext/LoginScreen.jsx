import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = (props) => {
  const { setUser } = useContext(UserContext);
  const handleLogIn = () => {
    setUser({
      id: 123,
      name: 'Alfonso Sempoalt',
    });
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button className='btn btn-primary mx-5' onClick={handleLogIn}>
        Log in
      </button>
    </>
  );
};

export default LoginScreen;
