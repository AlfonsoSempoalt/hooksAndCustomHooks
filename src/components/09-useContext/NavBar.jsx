import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const NavBar = () => {
  return (
    <>
      <nav className='navbar navbar-expand-sm navbar-dark bg-dark'>
        <ul>
          <Link to='/' className='navbar-brand'>
            NavBar
          </Link>
          <section className='collapse navbar-collapse' id='navbarNavAltMarkup'>
            <div className='navbar-nav'>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./'
                  exact
                >
                  Home
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./about'
                  exact
                >
                  About
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./login'
                  exact
                >
                  Log in
                </NavLink>
              </li>
            </div>
          </section>
        </ul>
      </nav>
    </>
  );
};

export default NavBar;
