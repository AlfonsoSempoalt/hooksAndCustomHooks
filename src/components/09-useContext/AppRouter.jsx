import React from 'react';
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from 'react-router-dom';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';

const AppRouter = () => {
  return (
    <>
      <Router>
        <section>
          <NavBar />

          <Switch>
            <Route exact={true} path='/about' component={AboutScreen} />
            <Route exact={true} path='/login' component={LoginScreen} />
            <Route exact={true} path='/' component={HomeScreen} />
            <Redirect to='/' />
          </Switch>
        </section>
      </Router>
    </>
  );
};

export default AppRouter;
