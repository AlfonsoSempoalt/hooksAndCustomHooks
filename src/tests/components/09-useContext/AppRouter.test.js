import React from 'react';
import { mount } from 'enzyme';
import AppRouter from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Name of the group', () => {

    const user = {
        id: 123,
        name: 'Alfonso Sempoalt',
    };

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('should match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
