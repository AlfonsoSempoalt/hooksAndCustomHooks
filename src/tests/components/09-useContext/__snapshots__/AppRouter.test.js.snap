// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should match with the snapshot 1`] = `
<section>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <ul>
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        NavBar
      </a>
      <section
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <li>
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Log in
            </a>
          </li>
        </div>
      </section>
    </ul>
  </nav>
  Array [
    <h1>
      Home page
    </h1>,
    <hr />,
    <pre>
      {"id":123,"name":"Alfonso Sempoalt"}
    </pre>,
  ]
</section>
`;
